docs: update creating autoware repositories page (#446)
autowarefoundation/autoware-documentation: docs/how-to-guides/integrating-autoware/creating-your-autoware-repositories/creating-autoware-repositories.md

# Creating Autoware repositories

> ## What is Meta-repository?

Reviewer: ```suggestion
## What is a Meta-repository?
```

Author: done :+1:

> A meta-repository is a repository that manages multiple repositories, and [Autoware](https://github.com/autowarefoundation/autoware) is one of them.
> It serves as a centralized control point for referencing, configuring,
> and versioning other repositories.
> In order to do that,
> Autoware meta-repository includes [`autoware.repos`](https://github.com/autowarefoundation/autoware/blob/main/autoware.repos) file for managing multiple repositories.
> We will use [VCS](https://github.com/dirk-thomas/vcstool) tool (Version Control System) for handling .repos file.
> VCS provides us with the capability to import, export, and pull from multiple repositories.
> 
> We will use VCS for importing all necessary repositories in our workspace.
> Please follow the documentation for VCS and .repos file usage.

Reviewer: Using Grammarly and ChatGPT I came up with a couple fixes and changes to this paragraph to improve its grammar and try to make it's text more concise. What do you think?
```suggestion
A meta-repository is a repository that manages multiple repositories, and [Autoware](https://github.com/autowarefoundation/autoware) is one of them. It serves as a centralized control point for referencing, configuring, and versioning other repositories. To accomplish this, the Autoware meta-repository includes the [`autoware.repos`](https://github.com/autowarefoundation/autoware/blob/main/autoware.repos) file for managing multiple repositories.
We will use the [VCS](https://github.com/dirk-thomas/vcstool) tool (Version Control System) to handle the .repos file. VCS provides us with the capability to import, export, and pull from multiple repositories.
VCS will be used to import all the necessary repositories to build Autoware into our workspace. Please refer to the documentation for VCS and .repos file usage.
```

Author: done :+1:

## How to create and customize your autoware meta-repository

### 1. Create autoware meta-repository

If you want to integrate Autoware into your vehicle, the first step is to create an Autoware meta-repository.

> One easy way is to fork autoware repository and clone it.
> (For how to fork a repository, refer to [GitHub Docs](https://docs.github.com/en/get-started/quickstart/fork-a-repo))

Reviewer: ```suggestion
One easy way is to fork the Autoware repository and clone it. (For instructions on how to fork a repository, refer to [GitHub Docs](https://docs.github.com/en/get-started/quickstart/fork-a-repo))
```

Author: done :+1:

> - In order to do that,
>   please go to the [autoware](https://github.com/autowarefoundation/autoware) repository
>   and click the fork button.
>   For example,
>   we will integrate `tutorial_vehicle` for these guidelines
>   (If you set up multiple types of vehicles,
>   adding a suffix like `autoware.vehicle_A` or `autoware.vehicle_B` is recommended),
>   so our forking should be like this:
> 
> ![forking-autoware_repository.png](images/forking-autoware_repository.png)

Reviewer: ```suggestion
-In this guide, Autoware will be integrated into a `tutorial_vehicle` (Note: when setting up multiple types of vehicles, adding a suffix like `autoware.vehicle_A` or `autoware.vehicle_B` is recommended).
For the first step, please visit the [autoware](https://github.com/autowarefoundation/autoware) repository and click the fork button. The fork process should look like this:

![forking-autoware_repository.png](images/forking-autoware_repository.png)
```

Author: done :+1:

Then click "Create fork" button to continue. After that, we can clone our fork repository on our local system.

```bash
git clone https://github.com/YOUR_NAME/autoware.<YOUR-VEHICLE>.git
```

For example, it should be for our documentation:

```bash
git clone https://github.com/leo-drive/autoware.tutorial_vehicle.git
```

#### 1.1 Create vehicle individual repositories

> 
> For integrating autoware on your own individual vehicles, you need fork and modify the following repositories as well:

Reviewer: ```suggestion

To integrate Autoware into your individual vehicles, you need to fork and modify the following repositories as well:
```

Author: done :+1:

- [sample_sensor_kit](https://github.com/autowarefoundation/sample_sensor_kit_launch): This repository will be used for sensing launch files, their pipelines-organizations and sensor descriptions.
  Please fork and rename as autoware meta-repository. At this point, our forked repository name will be `tutorial_vehicle_sensor_kit_launch`.
- [sample_vehicle_launch](https://github.com/autowarefoundation/sample_vehicle_launch): This repository will be used for vehicle launch files, vehicle_descriptions and vehicle_model.
  Please fork and rename this repository as well. At this point, our forked repository name will be `tutorial_vehicle_launch`.

> - [autoware_individual_params](https://github.com/autowarefoundation/autoware_individual_params): This repository stores parameters that change depending on each vehicle (i.e. sensor calibrations). Please fork
>   and rename this repository as well, our forked repository name will be `tutorial_vehicle_individual_params`.

Reviewer: ```suggestion
- [autoware_individual_params](https://github.com/autowarefoundation/autoware_individual_params): This repository stores parameters that change depending on each vehicle (i.e. sensor calibrations). Please fork
  and rename this repository as well; our forked repository name will be `tutorial_vehicle_individual_params`.
```

Author: done :+1:

> - [autoware_launch](https://github.com/autowarefoundation/autoware_launch): This contains node configurations and their parameters for Autoware. Please fork
>   and rename this repository as previous forked repositories, our forked repository name will be `autoware_launch.tutorial_vehicle`.
> 

Reviewer: ```suggestion
- [autoware_launch](https://github.com/autowarefoundation/autoware_launch): This repository contains node configurations and their parameters for Autoware. Please fork and rename this repository as previously forked repositories; our forked repository name will be `autoware_launch.tutorial_vehicle`.

```

Author: done :+1:

> ### 2. Customize your autoware.repos for your environment
> 
> You need to customize `autoware.repos` for your own vehicle's Autoware
> (It is included under the forked Autoware meta-repository directory).
> The all necessary autoware repos (except calibration and simulator repos)
> information is included in this file.
> 
> Now we can add forked individual repositories to `autoware.repos` before the importing operation.

Reviewer: ```suggestion
### 2. Customize autoware.repos for your environment

You need to customize your `autoware.repos` to import your forked repositories. `autoware.repos` usually includes information for all necessary Autoware repositories (except calibration and simulator repositories). Therefore, your forked repositories should also be added to this file.
```

Author: done :+1:

> #### 2.1 Adding individual repos to autoware.repos
> 
> After the forking of all repositories, we can start adding these repos to autoware.repos file.
> In order to do that, you should open autoware.repos via any text editor.
> You need to update `sample_sensor_kit_launch`,
> `sample_vehicle_launch`,
> `autoware_individual_params` and `autoware launch` with your own individual repos.
> In example at this tutorial,
> the necessary changes for our forked `tutorial_vehicle` repos should be like this:

Reviewer: ```suggestion
#### 2.1 Adding individual repos to autoware.repos

After forking all repositories, you can start adding them to your Autoware meta-repository by opening the `autoware.repos` file using any text editor and updating `sample_sensor_kit_launch`, `sample_vehicle_launch`, `autoware_individual_params` and `autoware launch` with your own individual repos.
For example, in this tutorial, the necessary changes for our forked `tutorial_vehicle` repositories should be as follows:
```

Author: done :+1:

  + sensor_kit/tutorial_vehicle_sensor_kit_launch:
  +   type: git
  +   url: https://github.com/leo-drive/tutorial_vehicle_sensor_kit_launch.git
  +   version: main
  ```

- Vehicle Launch:

  ```diff
  - vehicle/sample_vehicle_launch:
  -   type: git
  -   url: https://github.com/autowarefoundation/sample_vehicle_launch.git
  -   version: main
  + vehicle/tutorial_vehicle_launch:
  +   type: git
  +   url: https://github.com/leo-drive/tutorial_vehicle_launch.git
  +   version: main
  ```

- Individual Params:

  ```diff
  - param/autoware_individual_params:
  -   type: git
  -   url: https://github.com/autowarefoundation/autoware_individual_params.git
  -   version: main
  + param/tutorial_vehicle_individual_params:
  +   type: git
  +   url: https://github.com/leo-drive/tutorial_vehicle_individual_params.git
  +   version: main
  ```

- Autoware Launch:

  ```diff
  - launcher/autoware_launch:
  -   type: git
  -   url: https://github.com/autowarefoundation/autoware_launch.git
  -   version: main
  + launcher/autoware_launch.tutorial_vehicle:
  +   type: git
  +   url: https://github.com/leo-drive/autoware_launch.tutorial_vehicle.git
  +   version: main
  ```

> Please do same similar changes on your own autoware.repos file.
> After the changing these repositories,
> we will be ready to use VCS for importing all necessary repositories to our autoware workspace.

Reviewer: ```suggestion
Please make similar changes to your own autoware.repos file. After making these changes, you will be ready to use VCS to import all the necessary repositories into your Autoware workspace.
```

Author: done :+1:

> Firstly, we will create src directory under the own autoware meta-repository directory:

Reviewer: ```suggestion
First, create a src directory under your own Autoware meta-repository directory:
```

Author: done :+1:

```bash
cd <YOUR-AUTOWARE-DIR>
mkdir src
```

> Then, we will import all necessary repositories via:

Reviewer: ```suggestion
Then, import all necessary repositories via:
```

Author: done :+1:

```bash
cd <YOUR-AUTOWARE-DIR>
vcs import src < autoware.repos
```

> After the running `vcs import` command,
> all autoware repositories will be cloned in `src` folder under the Autoware directory.

Reviewer: ```suggestion
After the running `vcs import` command, all autoware repositories will be cloned in the `src` folder under the Autoware directory.
```

Author: done :+1:

> So, we can build our own repository with colcon build:

Reviewer: ```suggestion
Now, you can build your own repository with colcon build:
```

```bash
cd <YOUR-AUTOWARE-DIR>
colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release
```

> Please refer to the following documentation link for instructions on how to create and customize each `vehicle_interface`:
> 

Reviewer: ```suggestion
Please refer to the following documentation links for instructions on how to create and customize each of your vehicle's packages:
```

Author: done :+1:
